### DISCUSSION TITLE
Pythonesque parallel syntax

### DISCUSSION THREAD

**Discussion by jwezel** [carbon-language/carbon-lang#2708] [Language design]
I wonder how popular the idea of a pythonesque _parallel_ syntax (similar to Vala/Genie) would be here.

**Comment by geoffromer** [comment]
Can you give some links or examples of the kind of syntax you mean?

**Reply by jwezel** [reply]
Well, the syntax of the [Python](https://www.python.org/) language. Like
```python
    def insert(self, value):
        """
        Insert the specified value into the BST.
        """
        # Get the sort key for this value.
        if self._sort_key is None:
            sort_key = value
        else:
            sort_key = self._sort_key(value)
        # Walk down the tree until we find an empty node.
        node = self._root
        while node:
            if sort_key < node[_SORT_KEY]:
                node = node[_LEFT]
            else:
                node = node[_RIGHT]
        # Put the value in the empty node.
        if sort_key is value:
            node[:] = [[], [], value]
        else:
            node[:] = [[], [], value, sort_key]
        self._len += 1
```

[Genie](https://wiki.gnome.org/Projects/Genie) has a similar syntax:
```Genie
class Test:Object
    delegate DelegateType (a : int)
    data : int = 5
    d : DelegateType

    construct()
        self.d = method

    def method( b:int )
        print "%d %d", b, data

    def run ( c:int )
        d( c )

init
    var t = new Test()
    t.run( 1 )
```

**Reply by geoffromer** [reply]
Can you be more specific? Those two examples don't seem like they have much in common, and I don't know what you mean by "parallel" in this context, so I can't tell what specific syntax features you're interested in.

**Comment by jwezel** [comment]
These are two examples of a kind of a syntax that is not based on `{` and `}` and instead implements block structure by indentation.

By "parallel" I mean the dual syntax of Vala and Genie. They are compiled by the same compiler that understands two syntaxes: [Vala](https://wiki.gnome.org/Projects/Vala) which looks more like Java/C/C++ and [Genie](https://wiki.gnome.org/Projects/Genie) that looks more like Python. I find the idea of having two syntaxes striking. One for the league of those who prefer C/C++/Java-like syntax and one for those who prefer a [Python](https://www.python.org)-like syntax. Two frontends, one backend.

**Reply by geoffromer** [reply]
Oh, I see. It's not clear to me what problems that would solve, or how it would align with [Carbon's goals](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/goals.md).
